### DISCUSSION TITLE
Set browser screen resolution

### DISCUSSION THREAD

**Discussion by oleksiivasylenko** [cefsharp/CefSharp#3463] [Q&A]
SETUP: `windows 10, c#, winforms app. Net 4.7.2`

I am trying to set browser screen resolution.
I've tried to change Size property of browser as well as the size of browser(control size).
Tried to set maxWidth of winform and tried to set via cef settings 
`chromeSettings.CefCommandLineArgs.Add("window-size", "800,600");`
or
`chromeSettings.CefCommandLineArgs.Add("window-height", "800");`
and nothing helped, I always see my resolution which was before settings. 
To check resolution I used [website](https://bestfirms.com/what-is-my-screen-resolution/).

If somebody knows how to do it, I will be really tanksful!

**Comment by amaitland** [comment]
Are you trying to emulate a different screen resolution? 

You should be able to do that using DevTools 

https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setDeviceMetricsOverride

You can directly use DevTools protocol see https://github.com/cefsharp/CefSharp/issues/3165
http://cefsharp.github.io/api/88.2.x/html/M_CefSharp_DevTools_Emulation_EmulationClient_SetDeviceMetricsOverrideAsync.htm

**Reply by oleksiivasylenko** [reply]
Yes, that's it, thank you very much!!

Could you please clarify why set resolution, works only when I set `mobile` to `true`?

```
var client = _browserForm._browser1.chromeBrowser.GetDevToolsClient();
await client.Emulation.SetDeviceMetricsOverrideAsync(1000, 800, 1, true);
```

but in this case resolution doesn't change
```
var client = _browserForm._browser1.chromeBrowser.GetDevToolsClient();
await client.Emulation.SetDeviceMetricsOverrideAsync(1000, 800, 1, false); 
```

**Reply by amaitland** [reply]
> Could you please clarify why set resolution, works only when I set mobile to true?

I've never used Emulation.setDeviceMetricsOverride before,  the API is generated programatically, I've only used a small portion of it. Looks like the parameters are being passed correctly.

**Reply by amaitland** [reply]
Does the change require a reload or navigation to take effect?

**Reply by oleksiivasylenko** [reply]
yes for me to change the resolution I did navigation to "google" and then back, perhaps reloading are also works but I haven't tried it.

**Reply by oleksiivasylenko** [reply]
> > Could you please clarify why set resolution, works only when I set mobile to true?
> 
> I've never used Emulation.setDeviceMetricsOverride before, the API is generated programatically, I've only used a small portion of it. Looks like the parameters are being passed correctly.

Maybe it's chrome related settings(things), but I think it should set the resolution without affecting `isMobile` property

**Reply by amaitland** [reply]
You've only passed in values for the width and height, you haven't specified the screenWidth or screenHeight parameters. Looking at the source and the defaults are different for desktop and mobile. 

https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/frame/screen_metrics_emulator.cc;l=60;drc=184bf7b03160ebf2217ab4711118d013a750908a
